lib/pihole/base.py

```
@@ -84,46 +62,58 @@
+        logger.debug(f"'{furl}' with '{pArgs}'")
+
+        try:
+            if method == "get":
+                response = self.sessions[pi].get(furl, timeout=(3.05, 5), verify=True)
+            else:
+                response = self.sessions[pi].post(furl, data=pArgs, timeout=(3.05, 5), verify=True)
+
+            response.raise_for_status()
+            return response.json()
+        except requests.exceptions.RequestException as e:
+            logger.error(f"API call failed: {e}")
```

Ruff (0.14.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)